packages/protocol/contracts-0.8/common/CeloUnreleasedTreasury.sol

```diff
@@ -46,11 +58,30 @@ contract CeloUnreleasedTreasury is UsingRegistry, ReentrancyGuard, Initializable
    * @param amount The amount to release.
    */
   function release(address to, uint256 amount) external onlyEpochManager {
-    require(address(this).balance >= amount, "Insufficient balance.");
+    if (!hasAlreadyReleased) {
+      remainingBalanceToRelease = address(this).balance;
+      hasAlreadyReleased = true;
+    }
+
+    require(remainingBalanceToRelease >= amount, "Insufficient balance.");
     require(getCeloToken().transfer(to, amount), "CELO transfer failed.");
+    remainingBalanceToRelease -= amount;
```

Comment:
can we use CEI pattern and move `remainingBalanceToRelease -= amount;` above getCeloToken().transfer ?